app/connectors/connectors/jms/activemq-connectors/activemq-model/src/main/java/io/syndesis/connector/jms/ActiveMQUtil.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.util.ObjectHelper;
+
+/**
+ * Utility class for ActiveMQ Connection factory configuration.
+ *
+ * @author dhirajsb
+ */
+public class ActiveMQUtil {
+
+    public static KeyManager[] createKeyManagers(String clientCertificate) throws GeneralSecurityException, IOException {
+        final KeyStore clientKs = createKeyStore("amq-client", clientCertificate);
+
+        // create Key Manager
+        KeyManagerFactory kmFactory = KeyManagerFactory.getInstance("PKIX");
+        kmFactory.init(clientKs, null);
+        return kmFactory.getKeyManagers();
+    }
+
+    public static TrustManager[] createTrustManagers(String brokerCertificate) throws GeneralSecurityException,
+            IOException {
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("X509");
+        trustManagerFactory.init(createKeyStore("amq-server", brokerCertificate));
+        return trustManagerFactory.getTrustManagers();
+    }
+
+    private static KeyStore createKeyStore(String alias, String certificate) throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+        final KeyStore keyStore = KeyStore.getInstance("JKS");
+        keyStore.load(null, null);
+        // create client key entry
+        CertificateFactory factory = CertificateFactory.getInstance("X.509");
+        final Certificate generated = factory.generateCertificate(new ByteArrayInputStream
+                (certificate.getBytes("UTF-8")));
+        keyStore.setCertificateEntry(alias, generated);
+        return keyStore;
+    }
+
+    public static ActiveMQConnectionFactory createActiveMQConnectionFactory(String brokerUrl, String username, String
+            password, String brokerCertificate, String clientCertificate) {
+        final ActiveMQConnectionFactory connectionFactory;
+        if (brokerUrl.contains("ssl")) {
+            if (ObjectHelper.isEmpty(username)) {
+                connectionFactory = new ActiveMQSslConnectionFactory(brokerUrl);
+            }
+            else {
+                connectionFactory = new ActiveMQSslConnectionFactory(brokerUrl);
+                connectionFactory.setUserName(username);
+                connectionFactory.setPassword(password);
+            }
+
+            try {
+                // create client key manager
+                final KeyManager[] keyManagers;
+                if (ObjectHelper.isEmpty(clientCertificate)) {
+                    keyManagers = null;
+                } else {
+                    keyManagers = createKeyManagers(clientCertificate);
+                }
+
+                // create client trust manager
+                final TrustManager[] trustManagers;
+                if (ObjectHelper.isEmpty(brokerCertificate)) {
+                    // use a trust all TrustManager
+                    trustManagers = new TrustManager[] {new TrustAllTrustManager()};
```

Comment:
Perhaps we should go with the default TrustManager, i.e. only invoke `setKeyAndTrustManagers` if `brokerCertificate` is given?

Reply:
no, we still need to set a special TrustAllManager, or it won't connect.